src/Unleash/DefaultUnleash.cs

```diff
@@ -155,9 +160,43 @@ public bool IsEnabled(string toggleName, UnleashContext context, bool defaultSet
                         GetStrategyOrUnknown(s.Name)
                         .IsEnabled(s.Parameters, enhancedContext, ResolveConstraints(s).Union(s.Constraints))
                     );
+            }
+
+            if (featureToggle.Dependencies.Any() && !ParentDependenciesAreSatisfied(featureToggle, enhancedContext))
+            {
+                return false;
+            }
+
+            return strategy != null;
+        }
+
+        private bool ParentDependenciesAreSatisfied(FeatureToggle featureToggle, UnleashContext context)
+        {
+            return featureToggle.Dependencies.All(d => DependenciesSatisfied(d, context));
+        }
+
+        private bool DependenciesSatisfied(Dependency dependency, UnleashContext context)
+        {
+            var parentToggle = GetToggle(dependency.Feature);
+            if (parentToggle == null)
+            {
+                warnOnce.Warn(dependency.Feature, $"UNLEASH: Parent feature toggle {dependency.Feature} was not found in the cache, the evaluation of this dependency will always be false");
```

Comment:
This will log once per parent that's disabled. So if you have toggle A and toggle B that both depend on toggle C, which is missing, once A is evaluated and this message shows,  the B will never get a warning message. Is that what we want? (Genuinely asking, I've heard about warn once but this is the first time I've seen it)

Reply:
https://github.com/Unleash/unleash-client-node/pull/520

In the intro text:
> warning events for missing dependencies are reported once
> `this.warnedDependencies[missingDependency + name] = true;`

So no, should warn once per child and dependency it seems, I'll fix

Reply:
I've addressed this now